python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -1158,6 +1160,176 @@ def f():
       .isNotNull();
   }
 
+  @Test
+  void inferLoopVarType() {
+    var root = inferTypes("""
+      def f():
+        l = [1,2,3]
+        for i in l:
+          i
+      """);
+
+    var iType = TreeUtils.firstChild(root.statements().statements().get(0), ExpressionStatement.class::isInstance)
+      .map(ExpressionStatement.class::cast)
+      .flatMap(expressionStatement -> TreeUtils.firstChild(expressionStatement, Name.class::isInstance))
+      .map(Name.class::cast)
+      .map(Expression::typeV2)
+      .get();
+
+    Assertions.assertThat(iType).isInstanceOf(ObjectType.class)
+      .extracting(PythonType::unwrappedType)
+      .isInstanceOf(ClassType.class)
+      .extracting(PythonType::name)
+      .isEqualTo("int");
```

Comment:
You can have a direct type comparison to `INT_TYPE` instead of this indirect check (same for the other tests).